File: Sources/RTLSCore/LocationStore.swift
import Foundation

public protocol LocationStore: Sendable {
    func insert(points: [LocationPoint]) async throws

    func fetchPendingPoints(limit: Int) async throws -> [LocationPoint]
    func pendingCount() async throws -> Int
    func oldestPendingRecordedAt() async throws -> Date?

    func markSent(pointIds: [UUID], sentAt: Date) async throws
    func markFailed(pointIds: [UUID], errorMessage: String) async throws
}

File: rtls-kmp/src/commonMain/kotlin/com/rtls/kmp/LocationStore.kt
package com.rtls.kmp

interface LocationStore {
    suspend fun insert(points: List<LocationPoint>)
    suspend fun fetchPendingPoints(limit: Int): List<LocationPoint>
    suspend fun pendingCount(): Int
    suspend fun oldestPendingRecordedAt(): Long?
    suspend fun markSent(pointIds: List<String>, sentAtMs: Long)
    suspend fun markFailed(pointIds: List<String>, errorMessage: String)
}
